Mathlib/Topology/Baire/Lemmas.lean

```diff
@@ -47,6 +48,41 @@ theorem dense_iInter_of_isOpen_nat {f : ℕ → Set X} (ho : ∀ n, IsOpen (f n)
     (hd : ∀ n, Dense (f n)) : Dense (⋂ n, f n) :=
   BaireSpace.baire_property f ho hd
 
+/-- If `p : Y → X` is an open embedding and `X` is a Baire space, then `Y` is a Baire space. -/
+theorem Topology.IsOpenEmbedding.baireSpace {Y : Type*} [TopologicalSpace Y] {p : Y → X}
+    (hp : Topology.IsOpenEmbedding p) : BaireSpace Y := by
+  constructor
+  intro f hof hdf
+  let s := range p
+  let c := fun n : ℕ => p '' f n ∪ (closure s)ᶜ
+  have c_open (n : ℕ) : IsOpen (c n) := IsOpen.union (hp.isOpenMap (f n) (hof n))
+    isClosed_closure.isOpen_compl
+  have c_dense (n : ℕ) : Dense (c n) := by
+    rw [dense_iff_closure_eq, subset_antisymm_iff]
+    have : univ ⊆ closure (c n) := calc
+      _ ⊆ (interior (closure s)) ∪ (interior (closure s))ᶜ := by grind
+      _ ⊆ closure s ∪ (interior (closure s))ᶜ := by gcongr; exact interior_subset
+      _ ⊆ closure (p '' f n) ∪ (interior (closure s))ᶜ := union_subset_union
+          (closure_minimal (hp.continuous.range_subset_closure_image_dense (hdf n))
+          isClosed_closure) (subset_refl (interior (closure s))ᶜ)
+      _ ⊆ closure (p '' f n ∩ s) ∪ (interior (closure s))ᶜ := by gcongr; simp [s]
```

Comment:
not sure why you intersect with `s` on this line, only to remove it on the next one.